packages/core/src/validate/run.ts

```diff
@@ -629,6 +661,129 @@ function* getInternalFunctionStorageErrors(
+): Generator<ValidationErrorInitializer> {
+  if (contractDef.baseContracts.length > 0) {
+    const baseContractDefs = contractDef.baseContracts.map(base =>
+      deref('ContractDefinition', base.baseName.referencedDeclaration),
+    );
+    const baseContractsInitializersMap = new Map(
+      baseContractDefs.map(base => [base.name, getPossibleInitializers(base, true)]),
+    );
+    const baseContractsWithInitializers = baseContractDefs
+      .filter(base => baseContractsInitializersMap.get(base.name)?.length)
+      .map(base => base.name);
+
+    if (baseContractsWithInitializers.length > 0) {
+      // Check for missing initializers
+      const contractInitializers = getPossibleInitializers(contractDef, false);
+      if (contractInitializers.length === 0 && !skipCheck('missing-initializer', contractDef)) {
+        yield {
+          kind: 'missing-initializer',
+          src: decodeSrc(contractDef),
+        };
+      }
+
+      for (const contractInitializer of contractInitializers) {
+        const uninitializedBaseContracts = [...baseContractsWithInitializers];
+        const calledInitializerIds: number[] = [];
+
+        const expressionStatements =
+          contractInitializer.body?.statements?.filter(stmt => stmt.nodeType === 'ExpressionStatement') ?? [];
+        for (const stmt of expressionStatements) {
+          const fnCall = stmt.expression;
+          if (
+            fnCall.nodeType === 'FunctionCall' &&
+            (fnCall.expression.nodeType === 'Identifier' || fnCall.expression.nodeType === 'MemberAccess')
+          ) {
+            const referencedFn = fnCall.expression.referencedDeclaration;
+
+            // If this is a call to a parent initializer, then:
+            // - Check for duplicates
+            // - Check if the parent initializer is called in the correct order
+            for (const [baseName, initializers] of baseContractsInitializersMap) {
+              const foundParentInitializer = initializers.find(init => init.id === referencedFn);
+              if (referencedFn && foundParentInitializer) {
+                const duplicate = calledInitializerIds.includes(referencedFn);
+                if (
+                  duplicate &&
+                  !skipCheck('duplicate-initializer-call', contractDef) &&
+                  !skipCheck('duplicate-initializer-call', contractInitializer) &&
+                  !skipCheck('duplicate-initializer-call', stmt)
+                ) {
+                  yield {
+                    kind: 'duplicate-initializer-call',
+                    src: decodeSrc(fnCall),
+                    parentInitializer: foundParentInitializer.name,
+                    parentContract: baseName,
+                  };
+                }
+                calledInitializerIds.push(referencedFn);
+
+                const index = uninitializedBaseContracts.indexOf(baseName);
+                if (
+                  !duplicate &&
+                  index !== 0 &&
```

Comment:
If a userz disables the `duplicate-initializer-call` and there's a duplicate, we won't be yielding the `incorrect-initializer-order` check, is this correct?

Perhaps defining `duplicate` as follows would work:

```solidity 
const duplicate = calledInitializerIds.includes(referencedFn) &&
                  !skipCheck('duplicate-initializer-call', contractDef) &&
                  !skipCheck('duplicate-initializer-call', contractInitializer) &&
                  !skipCheck('duplicate-initializer-call', stmt)
```

Reply:
This is intentional, because `incorrect-initializer-order` should only be for things that are not duplicate (regardless of whether the duplicate was ignored).

For example if parent contracts are `A, B, C`, and the initializer calls were in the order of `A, B, B, C`, the third call could be seen as both a duplicate and incorrect order, because it does not match the expected order.  But we want to ignore it from being considered incorrect order in this case.

Even if the user disables `duplicate-initializer-call`, I think this case should still not be considered as incorrect order.

Added some comments and a separate testcase to clarify this.

Reply:
I see what you mean, basically having a duplicate will already be considered out of order. So if a user disables `duplicate-initializer-call`, those would be out of order anyways.

Makes sense. Thanks for documenting further. Solving